### ISSUE TITLE
How to judge whether the model converge?

### ISSUE THREAD

**Issue by lxtGH** [ClementPinard/SfmLearner-Pytorch#15]
Thanks for your code, I don't know how to judge the model converge when traning on cityscape dataset.
loss? or just after 3000 epoch

**Comment by ClementPinard** [comment]
Hello,

Hard to say ! It still remains an open problem as photometric loss doesn't exactly reflect depth map quality. As you can see in original implementation (https://github.com/tinghuiz/SfMLearner/issues/42), even for KITTI convergence is not perfect science, as you just "notice overfitting" by evaluating depth with GT at different steps.

I tried to make this search easier by adding GT values to validation steps in the very training, that way you don't have to randomly stop training and test it, but trigger automatically an "early stop", but for that you need GT, which is not avalaible in Cityscapes.

The main purpose of CS is only to pretrain the network and let him generalize better, there's no way to measure depth quality on CityScapes
